ai/unet/Model Training.py

```
@@ -0,0 +1,528 @@
+            raise ValueError(f"No images found in {images_dir}!")
+        print(f"  Found {len(self.image_files)} images")
+
+        missing = [f.name for f in self.image_files[:10]
+                   if not (self.masks_dir / f"{f.stem}.png").exists()]
+        if missing:
+            print(f"  WARNING: Missing masks for: {missing}")
+
+    def __len__(self):
+        return len(self.image_files)
+
+    def __getitem__(self, idx):
```

Ruff (0.15.7)

[warning] Missing return type annotation for special method `__getitem__`

(ANN204)